app/src/main/kotlin/com/flxrs/dankchat/main/MainViewModel.kt

```diff
@@ -144,7 +145,10 @@ class MainViewModel @Inject constructor(
 
     val suggestions: Flow<List<Suggestion>> =
         combine(emoteSuggestions, userSuggestions, supibotCommandSuggestions) { emoteSuggestions, userSuggestions, supibotCommandSuggestions ->
-            userSuggestions + supibotCommandSuggestions + emoteSuggestions
+            if (preferEmoteSuggestions.value)
```

Comment:
Add the new flow to the combine function so it can detect when it is set to a new value:
```kt
val suggestions: Flow<List<Suggestion>> =
        combine(emoteSuggestions, userSuggestions, supibotCommandSuggestions, preferEmoteSuggestions) { emoteSuggestions, userSuggestions, supibotCommandSuggestions, preferEmoteSuggestions ->
            if (preferEmoteSuggestions) {
                (emoteSuggestions + userSuggestions + supibotCommandSuggestions)
            } else {
                (userSuggestions + emoteSuggestions + supibotCommandSuggestions)
            }
        }
```

Also when using if statements I prefer them with braces. Alternatively you could use a when statement:
```kt
when {
    preferEmoteSuggestions -> (emoteSuggestions + userSuggestions + supibotCommandSuggestions)
    else -> (userSuggestions + emoteSuggestions + supibotCommandSuggestions)
}
```